src/app/components/utility/modal/modal.component.ts

```diff
@@ -230,33 +235,75 @@ export class ModalComponent implements OnInit {
     this.denyCallback();
   }
 
-  validateModalInput(e) {
-    this.inputErrorMessage = '';
-    if (e.target.name === 'challegenDeleteInput') {
-      this.isDisabled = e.target.value !== this.challenge.title;
-    } else if (e.target.name === 'editChallengeTitle') {
-      this.isDisabled = e.target.value === this.challenge.title;
-    } else if (e.target.name === 'update_first_name') {
-      this.isDisabled = e.target.value === this.user.first_name;
-    } else if (e.target.name === 'update_last_name') {
-      this.isDisabled = e.target.value === this.user.last_name;
-    } else if (e.target.name === 'update_affiliation') {
-      this.isDisabled = e.target.value === this.user.affiliation;
-    } else if (e.target.name === 'old_password') {
-      this.oldPassword = e.target.value;
-    } else if (e.target.name === 'new_password1') {
-      this.newPassword = e.target.value;
-      if (e.target.value === this.oldPassword) {
-        this.inputErrorMessage = 'Old password cannot be same as New Password';
-      }
-    } else if (e.target.name === 'new_password2') {
-      this.retype_newPassword = e.target.value;
-      if (e.target.value !== this.newPassword) {
-        this.inputErrorMessage = 'Password do not match';
-      }
-    }
+  validURL(string) { // https://stackoverflow.com/a/49849482/10103199
+    const res = string.match(/(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/g);
+    return (res !== null);
   }
 
+   validateModalInput(e) {
+
+     this.inputErrorMessage = '';
+     if (e.target.name === 'challegenDeleteInput') {
+       this.isDisabled = e.target.value !== this.challenge.title;
+     } else if (e.target.name === 'editChallengeTitle') {
+       this.isDisabled = e.target.value === this.challenge.title;
+     } else if (e.target.name === 'update_first_name') {
+       this.isDisabled = e.target.value === this.user.first_name;
+     } else if (e.target.name === 'update_last_name') {
+       this.isDisabled = e.target.value === this.user.last_name;
+     } else if (e.target.name === 'update_affiliation') {
+       this.isDisabled = e.target.value === this.user.affiliation;
+     } else if (e.target.name === 'update_google_scholar_url') {
+       if (this.validURL(e.target.value)) {
```

Comment:
You should also add condition for empty, these url fields are not required so a user can update his profile without entering the url fields.

Reply:
I don't think there's a need for an empty condition? The user can still empty their field even though an error would pop up.

Reply:
yeah but you need to hide the error for that empty field, currently it is displaying the error text.